src/Data/Text/Lazy/IO.hs

```diff
@@ -133,7 +133,9 @@ hPutStr h = mapM_ (T.hPutStr h) . L.toChunks
 
 -- | Write a string to a handle, followed by a newline.
 hPutStrLn :: Handle -> Text -> IO ()
-hPutStrLn h t = hPutStr h t >> hPutChar h '\n'
+hPutStrLn h Empty = hPutChar h '\n'
+hPutStrLn h (Chunk t Empty) = T.hPutStrLn h t  -- print the newline after the last chunk atomically
+hPutStrLn h (Chunk t ts) = T.hPutStr t >> hPutStrLn h ts
```

Comment:
> I think a simpler way is to decompose `Text.hPutStr` into `\h -> Stream.hPutStr h . Text.stream` (for some `Stream.hPutStr :: Handle -> Stream Char -> IO ()`) so you get `Lazy.hPutStr h = Stream.hPutStr h . Lazy.stream`.

`hPutStr` and `hPutStrLn` are not using the `Stream.hPutStr` mentioned in #592 . This under utilizes the character buffer for block buffering. They should be defined with `hPutStream` from Fusion.

Reply:
Oh yeah that was before Bodigrim asked me to submit a simpler version first https://github.com/haskell/text/pull/600#discussion_r1619495047